buildSrc/src/main/kotlin/io/spine/internal/gradle/javascript/task/Assemble.kt

```diff
@@ -0,0 +1,181 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.internal.gradle.javascript.task
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.node.ObjectNode
+import io.spine.internal.gradle.base.assemble
+import org.gradle.api.Task
+import org.gradle.api.tasks.TaskContainer
+import org.gradle.api.tasks.TaskProvider
+
+/**
+ * Registers tasks for assembling JavaScript artifacts.
+ *
+ * List of tasks to be created:
+ *
+ *  1. [TaskContainer.assembleJs];
+ *  2. [TaskContainer.compileProtoToJs];
+ *  3. [TaskContainer.installNodePackages];
+ *  4. [TaskContainer.updatePackageVersion].
+ *
+ * An example of how to apply it in `build.gradle.kts`:
+ *
+ * ```
+ * import io.spine.internal.gradle.javascript.javascript
+ * import io.spine.internal.gradle.javascript.task.assemble
+ *
+ * // ...
+ *
+ * javascript {
+ *     tasks {
+ *         assemble()
+ *     }
+ * }
+ * ```
+ */
+fun JsTasks.assemble() {
+
+    installNodePackages()
+    compileProtoToJs()
+    updatePackageVersion()
+
+    assembleJs().also {
+        assemble.configure {
+            dependsOn(it)
+        }
+    }
+
+}
+
+
+/**
+ * Locates `assembleJs` task in this [TaskContainer].
+ *
+ * It is a lifecycle task that produces consumable JavaScript artifacts.
+ */
+val TaskContainer.assembleJs: TaskProvider<Task>
+    get() = named("assembleJs")
+
+private fun JsTasks.assembleJs() =
+    register("assembleJs") {
+
+        description = "Assembles JavaScript sources into consumable artifacts."
+        group = jsAssembleTask
+
+        dependsOn(
+            installNodePackages,
+            compileProtoToJs,
+            updatePackageVersion
+        )
+    }
+
+
+/**
+ * Locates `compileProtoToJs` task in this [TaskContainer].
+ *
+ * The task compiles Protobuf messages into JavaScript.
+ *
+ * This is a lifecycle task that performs no action itself. It is used to aggregate other tasks
```

Comment:
Let's settle down. Either it "performs no action itself" or  "compiles Protobuf messages".

Right now the whole text looks a copy-paste mistake. Please adjust.